usdigitalresponse/usdr-gost
# feat: add ability to notify users on report failure

### Ticket #2526
## Description
- When the audit report or treasury report generation fails, these changes ensure that the user that kicked-off the process receives an email notification regarding the failure.

## Screenshots / Demo Video
```
root@d73736a46890:/tmp/localstack/state/ses# cat oikyvayjefcrejhj-bswbiifc-hdyw-fohy-ectw-zvcwcbxktqwe-rgpcwu.json 
{"Id": "oikyvayjefcrejhj-bswbiifc-hdyw-fohy-ectw-zvcwcbxktqwe-rgpcwu", "Region": "us-west-2", "Destination": {"ToAddresses": ["asridhar@usdigitalresponse.org"], "CcAddresses": ["grants-helpdesk@usdigitalresponse.org"]}, "Source": "grants-identification@usdigitalresponse.org", "Subject": "Treasury Report generation has failed for USDR Tenant", "Body": {"text_part": "There was an error generating a your requested Treasury Report. \n                        Someone from USDR will reach out within 24 hours to debug the problem. \n                        We apologize for any inconvenience.", "html_part": "There was an error generating a your requested Treasury Report. \n                        Someone from USDR will reach out within 24 hours to debug the problem. \n                        We apologize for any inconvenience."}, "Timestamp": "2024-02-06T21:05:39"}root@d73736a46890:/tmp
root@d73736a46890:/tmp/localstack/state/ses# 

```

## Testing
Added new unit tests and updated existing ones

### Automated and Unit Tests

### Manual tests for Reviewer

- Ensure localstack is running and `ARPA_TREASURY_REPORT_SQS_QUEUE_URL="http://sqs.us-west-2.localhost.localstack.cloud:4566/000000000000/arpa-treasury-queue"` environment variable is setup
- Ensure localstack has a queue that is appropriately setup - `awslocal sqs create-queue --queue-name arpa-treasury-queue --region us-west-2`
- Add the following line after line 1098 in `server/src/arpa_reporter/services/generate-arpa-report.js`
```
throw new Error('test unhandled error');
```
- Go to `localhost:8080/arpa_reporter` and hit the "Send Audit Report" button.
- Verify that the email was generated in `./volume/tmp/localstack/state/ses/.` in the local machine or in `/tmp/localstack/state/ses` from within the localstack/localstack container.

## Checklist
